report/3.Theory/LanguageSpecification.tex

```diff
@@ -135,31 +164,26 @@ \section{Language Specification}
 If two numbers were passed, the obvious result should be the sum of the two numbers.
 If on the other hand two strings were passed, the function could be implemented in such a way that the result would be the concatenation of the two strings.
 
-The function overloading in HCL is however a bit limited in that functions can not be overloaded with different amounts of parameters, only different parameter types. 
-This is due to the nature of the syntax in HCL where function parameters are not directly coupled to a function. 
-Therefore it would be impossible for the compiler to determine what is parameters and what is functions.
-Consider this example:
-
-\begin{lstlisting}
-1 to 10 sum print
-\end{lstlisting}
-
-Obviously the sum function accepts a list that should be summed. 
-However, if the sum function could potentially also accept a modifier function, that were to alter the result of the sum function, it would be impossible for the compiler to determine whether print was that modifier function, or the sum should be passed as a parameter to the print function.
+Function overloading in HCL is limited in that functions can not be overloaded with same parameters but different return types. 
+Functions can only be overloaded to return different types, if the types of the parameters vary.
+Furthermore overloading with different amounts of parameters are also not possible in HCL.
+This is because of multiple reasons, one being simplicity for the user, not accidentally adding an unexpected parameter to a function call. 
+Another being simplicity in the parsing of the function parameters. 
+Since the ability to overload with different amount of parameters brings next to nothing to the language, it was decided that it was not worth the effort to implement.
```

Comment:
write something on the lines of that functions can have varying length of parameters by usage of tuples or lists.

For instance if i want to either use 2 or 3 elements i could make a function like: "(num x, num y): num" AND another one like "(num x, tuple<num,num>): num".. which also adds required brackets to make it more easily readable for some cases etc etc.